xwiki-rendering-macros/xwiki-rendering-macro-box/src/main/java/org/xwiki/rendering/macro/box/AbstractBoxMacro.java

```diff
@@ -284,12 +287,14 @@ public Block build() throws MacroExecutionException
                 // we add the title, if there is one
                 if (!StringUtils.isEmpty(titleParameter)) {
                     // Don't execute transformations explicitly. They'll be executed on the generated content later on.
-                    List<? extends Block> titleBlock = AbstractBoxMacro.this.contentParser.parse(
+                    List<Block> titleContentBlock = AbstractBoxMacro.this.contentParser.parse(
                         titleParameter, context, false, true).getChildren();
-
-                    // put the right metadata around it
-                    ret.addChildren(Collections.singletonList(new MetaDataBlock(titleBlock,
-                        AbstractBoxMacro.this.getNonGeneratedContentMetaData("title"))));
+                    // Put metadata around it so that it's inplace editable
+                    List<Block> titleMetadata = Collections.singletonList(new MetaDataBlock(titleContentBlock,
```

Comment:
Small improvement, slightly shorter and based on java APIs:
```suggestion
                    List<Block> titleMetadata = List.of(new MetaDataBlock(titleContentBlock,
```

Reply:
:+1: Thanks! 
This function did look odd, but I just reused what was already working to avoid breaking something unexpectedly :)